rethinkdb/datadog_checks/rethinkdb/config.py

```diff
@@ -0,0 +1,115 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import absolute_import
+
+from typing import Callable, Iterator, List, Optional
+
+import rethinkdb
+
+from datadog_checks.base import ConfigurationError
+
+from .connections import Connection, RethinkDBConnection
+from .exceptions import CouldNotConnect
+from .metrics.config import collect_config_totals
+from .metrics.current_issues import collect_current_issues
+from .metrics.statistics import (
+    collect_cluster_statistics,
+    collect_replica_statistics,
+    collect_server_statistics,
+    collect_table_statistics,
+)
+from .metrics.statuses import collect_server_status, collect_table_status
+from .metrics.system_jobs import collect_system_jobs
+from .queries import QueryEngine
+from .types import Instance, Metric
+from .version import parse_version
+
+
+class Config(object):
+    """
+    Hold instance configuration for a RethinkDB check.
+
+    Encapsulates the validation of an `instance` dictionary while improving type information.
+    """
+
+    def __init__(self, instance):
+        # type: (Instance) -> None
+        host = instance.get('host', 'localhost')
+        port = instance.get('port', 28015)
+        user = instance.get('user')
```

Comment:
can we use `username` for consistency with standard config option names?

Reply:
Ah yes, nice catch! Addressed in b8246a0.